src/components/blocks/Slug.vue

```diff
@@ -0,0 +1,21 @@
+<!-- Input field for slug management. -->
+<script setup lang="ts">
+const props = defineProps<{
+	request: Partial<{ slug: string }>
```

Comment:
Surely `{slug?: string}` is a better way to write this. I don't think this property should be optional though. Also, if you need to "write back" to any passed-in props, I believe you should be using `defineModel()` instead. See https://vuejs.org/guide/components/v-model.html.

Reply:
I didn't know about `defineModel`. This is good, as the reason why I wrote the prop type that way was to handle parent components with conflicting types (although I just realised I could've defined an interface type). I'll try to rewrite this bit with `defineModel`.